### PR TITLE
Implement ICAT

### LINKED ISSUE #211: Rework Intake Pivot/Climber from ground up
Take the intake. Take the climber. Burn them in a FIRE. This includes robot map and robot constants and robot preferences.
Make a commit, make a draft pr RIGHT HERE.
Then, begin the rework

### PR DESCRIPTION
**Pull request by dunep**

### CODE REVIEW

**Comment by alicekuznetsov** [comment]
We wanted both positional and velocity control on the climber. We also wanted them both to be Motion Magic. I'm not sure how we'll make sure that the PID works in both cases. Maybe it'll just work like magic? Maybe its not actually a problem and I'm just really tired?

**Comment by alicekuznetsov** [comment]
@TaylerUva ^^

**Comment by alicekuznetsov** [comment]
Also im trying to think about how we should account for other commands interrupting the PrepAmp command. As i have it written right now, everything kinda explodes if it fails to transfer the amp note into the amper and that would run until its interrupted and them maybe things will explode and idk I'm very tired

**Comment by TaylerUva** [comment]
> We wanted both positional and velocity control on the climber. We also wanted them both to be Motion Magic. I'm not sure how we'll make sure that the PID works in both cases. Maybe it'll just work like magic? Maybe its not actually a problem and I'm just really tired?

You would have 2 PIDs but we don't need velocity in climber just voltage

**Comment by alicekuznetsov** [comment]
TESTING NOTES
- [x] CAN ids
- [x] Set value of Intake ABS_ENCODER_OFFSET and Intake ABS_ENCODER_INVERT 
- [x] Determine if we will need to add an ABS_ENCODER_ROLLOVER
- [x] Probably comment out stowing the pivot for now
- [x] Set software limits

**Comment by TaylerUva** [comment]
DISABLE CURRENT LIMITING ON downwards only

**Comment by TaylerUva** [comment]
Climber up waits for amp

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -69,11 +71,13 @@ public void disabledExit() {
   @Override
   public void autonomousInit() {
     m_autonomousCommand = m_robotContainer.getAutonomousCommand();
-    m_robotContainer.clearSubsystemMovements().schedule();
 
+    RobotContainer.zeroClimber().schedule();
+    // RobotContainer.stowIntakePivot().schedule();
```
Comment (TaylerUva) [inline src/main/java/frc/robot/Robot.java:76]: Make it check if we are less than stow angle

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -89,7 +93,11 @@ public void teleopInit() {
     if (m_autonomousCommand != null) {
       m_autonomousCommand.cancel();
     }
-    m_robotContainer.clearSubsystemMovements().schedule();
+
+    if (!hasAutonomousRun) {
+      RobotContainer.zeroClimber().schedule();
+      // RobotContainer.stowIntakePivot().schedule();
```
Comment (TaylerUva) [inline src/main/java/frc/robot/Robot.java:99]: Make it check if we are less than stow angle

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -143,50 +146,46 @@ private void configureDriverBindings(SN_XboxController controller) {
...
-    // subClimber)
-    // .alongWith(Commands.runOnce(() -> subTurret.setTurretAngle(0, false))));
-    // controller.btn_RightTrigger.onFalse(
-    // Commands.run(() -> subClimber.setClimberMotorSpeed(0), subClimber));
+    controller.btn_LeftTrigger
+        .whileTrue(
+            Commands.either(
+                new PrepAmp(subIntake, subPitch, subTransfer, subTurret, subShooter),
+                Commands.run(() -> subClimber.setPercentOutput(prefClimber.climberUpSpeed.getValue())),
+                () -> getLockedLocation() != LockedLocation.AMP).repeatedly())
```
Comment (TaylerUva) [inline src/main/java/frc/robot/RobotContainer.java:154]: May be SOURCE of jitter

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -143,50 +146,46 @@ private void configureDriverBindings(SN_XboxController controller) {
...
+        .onFalse(Commands.runOnce(() -> subClimber.setPercentOutput(0)));
+
+    controller.btn_RightTrigger
+        .onTrue(Commands.runOnce(() -> subClimber.setCurrentLimiting(false)))
+        .whileTrue(Commands.run(() -> subClimber.setPercentOutput(prefClimber.climberDownSpeed.getValue())))
+        .onFalse(Commands.run(() -> subClimber.setPercentOutput(0))
+            .alongWith(Commands.runOnce(() -> subClimber.setCurrentLimiting(false))));
```
Comment (TaylerUva) [inline src/main/java/frc/robot/RobotContainer.java:161]: This should be true

**Review by TaylerUva — APPROVED** [review]

**Review by TaylerUva — APPROVED** [review]

**Pull request merged**